sdk/platform/http_client/curl/src/az_curl.c

```diff
@@ -150,7 +150,7 @@ static AZ_NODISCARD az_result _az_http_client_curl_add_header_to_curl_list(
   az_span writable_buffer;
   {
     int32_t const buffer_size = az_span_size(header.key) + az_span_size(separator)
-        + az_span_size(header.value) + az_span_size(AZ_SPAN_FROM_STR("\0"));
+        + az_span_size(header.value) + 1 /*one for 0 terminated*/;
```

Comment:
nit: In other places, folks have preferred using `az_span_size` to be more self documenting rather than constants. We may want to converge and use the same for consistency.

https://github.com/Azure/azure-sdk-for-c/pull/568#discussion_r414133370

So, for now, I would leave things as is and revert this change. It is unrelated to the fix anyway.

Reply:
this was a @JeffreyRichter  request. He didn't like to see the az_span created just to calculate the size of it when we know it will be just 1.